ral-facilities/datagateway-api
# Implement Search API Endpoints

This PR will close #266, closes #267, closes #268

## Description
This PR implements all of the endpoints required for the search API. This is fairly simple code, it just utilises all of the stuff we've been working on - queries, filters, putting the data into a format ready to be given as a response. I've added docstrings where needed, which resolves some of the TODOs in `search_api_endpoints.py`. 

I have also added some tests for the new endpoints. As of the date of PR creation (14/01), these tests are fairly basic, but more tests will be added before this PR gets to be ready for review.

## Testing Instructions
Manually send requests to the endpoints and see if you can break them. As of the date of PR creation, they're fairly fragile because of bugs that need to be fixed elsewhere (which we know about already) but hopefully the endpoints should be robust by the time they're merged (or we have plans to fix bugs).

This branch will cause a feature version bump.
